lib/spack/spack/cmd/common/arguments.py

```diff
@@ -581,23 +581,51 @@ def add_concretizer_args(subparser):
 
 
 def add_connection_args(subparser, add_help):
-    subparser.add_argument(
-        "--s3-access-key-id", help="ID string to use to connect to this S3 mirror"
+    def add_argument_string_or_variable(parser, arg: str, *, deprecate_str: bool = True, **kwargs):
+        group = parser.add_mutually_exclusive_group()
+        group.add_argument(arg, **kwargs)
+        # Update help string
+        if "help" in kwargs:
+            kwargs["help"] = "Environment variable containing " + kwargs["help"]
```

Comment:
```suggestion
            kwargs["help"] = f"environment variable containing {kwargs['help']}"
```

minor but `e` for consistency